### DISCUSSION TITLE
Issue with Backtracking, using global data

### DISCUSSION THREAD

**Discussion by pkeys** [WAM2layers/WAM2layers#225] [Q&A] [answered]
Hi, 
I'm running the wam2layers, and I've recently encountered a persistent glitch that I'm not sure what the origin is. I've prepared this summary of the issue, along with some of the tests I've run. 

SUMMARY OF ISSUE
I am running the wam2layers (pip install version) on an Apple computer, using the Terminal. I am using era5 data at the 0.25x0.25 resolution, and running the data at the global scale (-180 to 180 longitude, -90 to 90 latitude). 

First, I have run the preprocessing script to calculate the fluxes and storages. Second, I have created a source region file for the Congo River basin, and have performed backtracking for the years 2020-2018. I have encountered a few persistent errors where the backtracking is working without error, and then there is a sudden explosion of negative (or positive) values that are completely spurious. 
![glitch](https://github.com/WAM2layers/WAM2layers/assets/30704657/bd2b8526-5263-4622-adaf-ee37dc07ce92)

One of these errors is for December 5, 2018. When I run the backtracking from Dec 17, 2018 or later, there are explosive errors in the data. When I run from Dec 16th 2018 or earlier, the errors do not occur. This suggests (possibly) that there is some sort of numerical issue that is accumulating in the backtracking data. 

METHODS OF TROUBLESHOOTING ISSUE
I have tried a number of approaches to solve this issue. 
- First, I redownloaded the era5 data, to ensure that there was not an error in my driving data. Then I re-ran the preprocessing scripts.  This did not solve the issue.
- Second, I re-ran the backtracking using the "periodic_boundary" set to "true" (as recommended for global data. This did not solve the issue. 
- Third, I have explored the sensitivity of the error to find out how many days of backtracking are required to replicate the issue. 
- Note, I have not yet tried to reduce the spatial domain of my data (i.e., not running all the way to the poles), though I am currently downloading that data right now. 

REQUEST
I am sharing this issue and the corresponding data in the hopes that you might be able to replicate the errors, and possibly shed some light on what might be happening. 

DATA 
In the [linked Google Drive folder](https://drive.google.com/drive/folders/1cOt4otlmiUdf49a2SaLt6CsWP2Eyly4-?usp=sharing), you will find the preprocessed data for the relevant days that you'll need to replicate the error (I was able to replicate the error on Dec 5 by beginning a backtracking run starting on Dec 17), along with the source region file (name = "congo_sample_source_region_updated.nc").

**Comment by ruudvdent** [comment]
Hi Pat, this is really odd behavior. We'll have a WAM2layers development week from 17-21 July. Hopefully we can have a more detailed look at this problem. First step I'll try to see if I can replicate the error.

**Reply by Peter9192** [reply]
@pkeys could you also share your configuration file?

**Comment by pkeys** [comment]
Hi Ruud. I'll keep troubleshooting on my end, but I look forward to hearing
what you are able to come up with.
Thank you,
-pat

On Fri, Jul 7, 2023 at 8:45 AM Ruud van der Ent ***@***.***>
wrote:

> Hi Pat, this is really odd behavior. We'll have a WAM2layers development
> week from 17-21 July. Hopefully we can have a more detailed look at this
> problem. First step I'll try to see if I can replicate the error.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/WAM2layers/WAM2layers/discussions/225#discussioncomment-6385460>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHKIIEMYFRTDFUYDAVEQSJTXPAOG7ANCNFSM6AAAAAAZQ7WGNE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

Dr. Patrick W. Keys | he, him, his
Assistant Professor | Dept. of Atmospheric Science | Colorado State
University
Email: ***@***.***  ***@***.***>
Website: http://patkeys.com/
Google: ***@***.***

**Comment by Peter9192** [comment]
Hi @pkeys thanks for the detailed description. Sounds like a numerical issue indeed. Have you tried setting a smaller time step? Also, since you mention that it happens specifically around a certain date, is there some sort of extreme event around that time with large gradients that could put an additional constraint on the stability?

**Comment by pkeys** [comment]
Thanks @Peter9192.
I re-ran the same time period with a reduced timestep (5min, instead of
10min), and the same results emerged (see figure):
[image: image.png]

I looked at the specific dates of the reanalysis data, and there doesn't
appear to be anything jumping out in terms of extreme events. I will keep
you posted as I keep troubleshooting on my end.

On Sun, Jul 9, 2023 at 6:33 AM Peter Kalverla ***@***.***>
wrote:

> Hi @pkeys <https://github.com/pkeys> thanks for the detailed description.
> Sounds like a numerical issue indeed. Have you tried setting a smaller time
> step? Also, since you mention that it happens specifically around a certain
> date, is there some sort of extreme event around that time with large
> gradients that could put an additional constraint on the stability?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/WAM2layers/WAM2layers/discussions/225#discussioncomment-6396030>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHKIIEI2KXR6MWKXMW2PTWDXPKQHPANCNFSM6AAAAAAZQ7WGNE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Dr. Patrick W. Keys | he, him, his
Assistant Professor | Dept. of Atmospheric Science | Colorado State
University
Email: ***@***.***  ***@***.***>
Website: http://patkeys.com/
Google: ***@***.***

**Comment by ruudvdent** [comment, marked as answer]
@pkeys. We've build in lot's a new logging and stability stuff in v3.0.0-beta6. Could you check if your problems are solved with that release?
It is now also possible to run subdomains, which comes in handy as you probably want to chop of anything above 75 degrees latitude
